ui/admin/app/routes/onboarding.js

```diff
@@ -1,36 +1,122 @@
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
-
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
+import { notifyError } from 'core/decorators/notify';
+import { TYPE_TARGET_TCP } from 'api/models/target';
 export default class OnboardingRoute extends Route {
   // =services
 
   @service store;
   @service session;
   @service router;
+  @service intl;
 
   // =methods
 
   /**
-   * If arriving here unauthenticated, redirect to index for further processing.
+   * If arriving here unauthenticated, or has orgs redirect to index for further processing.
    */
-  beforeModel() {
-    if (!this.session.isAuthenticated) this.router.transitionTo('index');
+  async beforeModel() {
+    const orgs = await this.store.query('scope', {
+      scope_id: 'global',
+    });
+    if (!this.session.isAuthenticated || orgs.length)
+      this.router.transitionTo('index');
   }
 
   /**
    * List all scopes (orgs) under global.
+   * Creare org, project, target and roles
    */
-  model() {
-    return this.store.query('scope', {
-      scope_id: 'global',
-    });
+  async model() {
+    return {
+      org: this.store.createRecord('scope', {
+        type: 'org',
+        scopeID: 'global',
+        name: 'SecOps',
+        description: 'Sample org created by quick setup',
+      }),
+      project: this.store.createRecord('scope', {
+        type: 'project',
+        name: 'AWS',
+        description: 'Sample project created by quick setup',
+      }),
+      target: this.store.createRecord('target', {
+        type: TYPE_TARGET_TCP,
+        name: 'Test target',
+        description: 'Sample target created by quick setup',
+      }),
+      role: this.store.createRecord('role', {
+        name: 'test_target_role',
+      }),
+    };
+  }
+
+  // =actions
+  @action
+  @loading
+  @notifyError(function () {
+    return this.intl.t('errors.onboarding-failed.description');
+  })
+  async createTarget(targetAddress, targetPort) {
+    // Create org and Project
+    try {
+      await this.createOrgAndProject();
+    } catch (error) {
```

Comment:
Why catch and rethrow a generic error? Would it make sense to separate org and project creation even further and if one fails, drop them off at either the org or project page depending on what failed?

Reply:
Changed, now it throws the error passed by param